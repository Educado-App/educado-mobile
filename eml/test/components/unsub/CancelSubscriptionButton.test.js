/**
 * This file contains the unit tests for the CancelSubscriptionButton component.
 * The component is tested for rendering and onPress function call.
 * @module CancelSubscriptionButtonTest
 */
import React from 'react';
import renderer from 'react-test-renderer';
import SubscriptionCancel from '../../../components/section/CancelSubscriptionButton';

/**
 * Describes the test suite for the SubscriptionCancel component.
 */
describe('<SubscriptionCancel />', () => {

  let pressed = false;

  /**
   * Mock function to set pressed to true when called.
   */
  const mockOnPress = () => {
    pressed = true;
  };

  let subscriptionCancel;

  /**
   * Resets modules and restores all mocks after all tests have run.
   */
  afterAll(async () => {
    jest.resetModules();
    jest.restoreAllMocks();
  });

  /**
   * Resets pressed to false and creates a new instance of SubscriptionCancel before each test.
   */
  beforeEach(async () => {
    pressed = false;
    subscriptionCancel = renderer.create(<SubscriptionCancel onPress={mockOnPress} />);
  });

  /**
   * Tests if the SubscriptionCancel button renders correctly.
   */
  it('SubscriptionCancel button renders', async () => {
    expect(await subscriptionCancel.toJSON()).toMatchSnapshot();
  });

  /**
   * Tests if the onPress function is called when the SubscriptionCancel button is pressed.
   */
  it('Pressing SubscriptionCancel button calls onPress function', async () => {
    const subscriptionCancelButton = await subscriptionCancel.root.findByProps({ testID: 'subscriptionCancelButton' });
    await renderer.act(() => {
      return subscriptionCancelButton.props.onPress();
    });
    expect(pressed).toBe(true);
  });
});