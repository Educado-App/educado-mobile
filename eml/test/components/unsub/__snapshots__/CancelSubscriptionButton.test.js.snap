// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionCancel /> SubscriptionCancel button renders 1`] = `
<View
  style={
    [
      {
        "paddingBottom": 16,
        "paddingTop": 16,
      },
      {
        "alignSelf": "center",
      },
      {
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#FF4949",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 4,
          },
        ]
      }
      testID="subscriptionCancelButton"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              [
                {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                {
                  "textTransform": "uppercase",
                },
                {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Cancelar inscrição
        </Text>
      </View>
    </View>
  </View>
</View>
`;
